extensions/cloudflare/src/view-sites.tsx

```diff
@@ -0,0 +1,172 @@
+        if (accounts.length === 0) {
+          setSites([]);
+          setLoading(false);
+        }
+        const account = accounts[0];
+        setAccountId(account.id);
+
+        const sites = await service.listZones(account);
+        setSites(sites);
+        setLoading(false);
+      } catch (e) {
+        setLoading(false);
+        handleNetworkError(e);
+      }
+    }
+
+    fetchSites();
+  }, []);
+
+  return (
+    <List isLoading={isLoading}>
+      {sites.map((site) => (
+        <List.Item
+          actions={
+            <ActionPanel>
+              <Action.Push
+                icon={Icon.TextDocument}
+                title="Show Details"
+                target={<SiteView accountId={accountId} id={site.id} />}
+              />
+              <Action.Push
+                icon={Icon.List}
+                title="Show DNS Records"
+                target={<DnsRecordView siteId={site.id} />}
+              />
+              <Action.OpenInBrowser
+                title="Open on Cloudflare"
+                url={getSiteUrl(accountId, site.name)}
+                shortcut={{ modifiers: ['cmd'], key: 'f' }}
+              />
+            </ActionPanel>
+          }
+          icon={getSiteStatusIcon(site.status)}
+          key={site.id}
+          title={site.name}
+        />
+      ))}
+    </List>
+  );
+}
+
+interface SiteProps {
+  accountId: string;
+  id: string;
+}
+
+function SiteView(props: SiteProps) {
+  const { accountId, id } = props;
+
+  const [site, setSite] = useState<Zone | null>(null);
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchSite() {
+      const site = await service.getZone(id);
+      setSite(site);
+      setLoading(false);
+    }
+
+    fetchSite();
+  }, []);
+
+  if (!site) {
+    return <Detail isLoading={isLoading} markdown="" />;
+  }
+
+  const markdown = `
+  # Site
+
+  ## Name
+
+  ${site.name}
+
+  ## Status
+
+  ${site.status}
+
+  ## Name servers
+
+  ${site.nameServers.map((server) => `* ${server}`).join('\n\n')}
+  `;
+  return (
+    <Detail
+      actions={
+        <ActionPanel>
+          <Action.Push
+            icon={Icon.List}
+            title="Show DNS Records"
+            target={<DnsRecordView siteId={site.id} />}
+          />
+          <Action.OpenInBrowser
+            title="Open on Cloudflare"
+            url={getSiteUrl(accountId, site.name)}
+            shortcut={{ modifiers: ['cmd'], key: 'f' }}
+          />
+        </ActionPanel>
+      }
+      isLoading={isLoading}
+      markdown={markdown}
+    />
+  );
+}
+
+interface DnsRecordProps {
+  siteId: string;
+}
+
+function DnsRecordView(props: DnsRecordProps) {
+  const { siteId } = props;
+
+  const [records, setRecords] = useState<DnsRecord[]>([]);
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    async function fetchRecords() {
+      const records = await service.listDnsRecords(siteId);
```

Comment:
`try/catch`?

Reply:
Fixed in 570669becea6763affe62d25b16e1b4d27798352.